datafusion/physical-plan/src/aggregates/row_hash.rs

```diff
@@ -859,14 +859,13 @@ impl GroupedHashAggregateStream {
                         )?;
                     }
                     _ => {
+                        if opt_filter.is_some() {
+                            return internal_err!("aggregate filter should be applied in partial stage, there should be no filter in final stage");
```

Comment:
My assumption

Reply:
Well the fact that no queries error and all tests pass is certainly compelling. I am not sure how good our test coverage is in this area, but this looks good to me